### PR TITLE
Code changes for serialization/deserialization

### PR DESCRIPTION
**Pull request by sarangan12**
This is a minor bug fix. Currently no code uses this. Please review and approve

### CODE REVIEW

File: runtime/ms_rest/lib/ms_rest/jsonable.rb [hunk 1]
```diff
@@ -13,8 +13,9 @@ module JSONable
     # @return [String] JSON serialized version of the object
     #
     def to_json(options = nil)
-      mapper = self.class.mapper if options.nil? || !options.key?(:mapper)
-      serialize(mapper, self)
+      mapper = (options.nil? || !options.key?(:mapper))? self.class.mapper: options[:mapper]
+      object = (options.nil? || !options.key?(:object))? self: options[:object]
```
Comment (veronicagg) [inline runtime/ms_rest/lib/ms_rest/jsonable.rb]: what would be passed in the ":object" option? What's the reason for introducing this? 
Would the result be proper if there's no ":mapper" option, but there's ":object" option passed?

**Comment by sarangan12** [comment]
@veronicagg @vishrutshah This is done to handle the issue https://github.com/Azure/azure-sdk-for-ruby/issues/610. 

In order to use the JSONable, we need to be able to pass both the object and mapper. Both are optional. You can pass one or both and the default values will be used. 

How will the resulting code look:

For composite models, it will be like:
```
Azure::ARM::Compute::Models::AvailabilitySet.to_json #Here you dont have to pass the mapper/object. 
```
For non composite models, it will be like:
```
def put_utc_max_date_time_async(datetime_body, custom_headers = nil)
..........
..........
request_mapper = {
        required: true,
        serialized_name: 'datetimeBody',
        type: {
          name: 'DateTime'
        }
      }
current_module = Object.const_get(self.class.to_s.split( '::' )[0, self.class.to_s.split( '::' ).length-1].join)
a_new_class = Class.new { include MsRest::JSONable }
current_module.const_set('ANewClass', a_new_class)
request_content = a_new_class.new.to_json({mapper: request_mapper, object: datetime_body })
..........
..........
```

**Comment by veronicagg** [comment]
@sarangan12 ok, should we be adding a test (or two) in ms_rest to check on this added functionality? It can help not only validate but describe the usage.

**Comment by sarangan12** [comment]
@veronicagg Why stop with 2? :-) added 3 cases 1 with mapper (without object), 1 with object (without mapper), 1 with object and mapper

**Pull request merged**